src/Features/CSharp/Portable/Completion/CompletionProviders/SymbolCompletionProvider.cs

```diff
@@ -99,22 +99,69 @@ internal override bool IsInsertionTrigger(SourceText text, int characterPosition
             return ValueTuple.Create(displayText, insertionText);
         }
 
-        private static CompletionItemRules s_importDirectiveRules =
-            CompletionItemRules.Create(commitCharacterRules: ImmutableArray.Create(CharacterSetModificationRule.Create(CharacterSetModificationKind.Replace, '.', ';')));
-        private static CompletionItemRules s_importDirectiveRules_preselect = s_importDirectiveRules.WithSelectionBehavior(CompletionItemSelectionBehavior.HardSelection);
+        protected override CompletionItemRules GetCompletionItemRules(List<ISymbol> symbols, SyntaxContext context, bool preselect)
+        {
+            return cachedRules[ValueTuple.Create(context.IsInImportsDirective, preselect,
+                ((CSharpSyntaxContext)context).IsPossibleTupleContext)] ??
```

Comment:
Indexing into an array will throw if the item is not in the dictionary.  you either need TryGetValue, or you remove the below optimization and you just include all 8 values.